pkg/volume/portworx/portworx_util.go

```diff
@@ -280,8 +280,31 @@ func createDriverClient(hostname string) (*osdclient.Client, error) {
 //            the Portworx node that will own/owns the data.
 func (util *PortworxVolumeUtil) getPortworxDriver(volumeHost volume.VolumeHost, localOnly bool) (volumeapi.VolumeDriver, error) {
 	var err error
+	kubeClient := volumeHost.GetKubeClient()
+	if kubeClient == nil {
+		err = fmt.Errorf("Failed to get kubeclient when creating portworx client")
+		glog.Errorf(err.Error())
+		return nil, err
+	}
+
+	opts := metav1.GetOptions{}
+	svc, err := kubeClient.CoreV1().Services(api.NamespaceSystem).Get(pxServiceName, opts)
```

Comment:
This issues API call for **each** portworx mount / unmount in kubelet. Can the port number change in a running cluster? Can be the port cached in any way?

Reply:
@jsafrane I've improved this in the incremental commit.

Now I use an existing saved client as long as it's valid. If port changes, a new client will be created and used subsequently.

I have also refactored the methods and created separate methods for getting a client from local calls (mount, unmount, attach and detach) vs cluster wide calls.